tests/MongoDB.Driver.Tests/Specifications/change-streams/ChangeStreamTestRunner.cs

```diff
@@ -280,19 +281,19 @@ private List<ChangeStreamDocument<BsonDocument>> ReadChangeStreamDocuments(IAsyn
         {
             var result = new List<ChangeStreamDocument<BsonDocument>>();
             var resultDocument = test["result"].AsBsonDocument;
-            if (!resultDocument.TryGetValue("success", out var successNode))
-            {
-                // skip an empty batch which is the result of an initial "aggregate" change stream request
-                // next `MoveNext` will call a server
-                cursor.MoveNext();
-            }
-            int expectedNumberOfDocuments = successNode?.AsBsonArray.Count ?? 0;
+            var successNode = resultDocument.GetValue("success", null)?.AsBsonArray;
 
+            var stopwatch = Stopwatch.StartNew();
             while (async ? cursor.MoveNextAsync().GetAwaiter().GetResult() : cursor.MoveNext())
             {
                 result.AddRange(cursor.Current);
 
-                if (result.Count >= expectedNumberOfDocuments)
+                if (successNode != null && result.Count >= successNode.Count)
+                {
+                    break;
+                }
+
+                if (stopwatch.Elapsed > TimeSpan.FromSeconds(10)) // this is enough time for the server to respond with non-empty batch if exists
```

Comment:
This comment is not accurate.

I would either remove the comment or make it more accurate, perhaps:

    // 10 seconds is enough time to receive all the required change documents or for an exception to be thrown

during those 10 seconds the server might return empy and non-empty batches interspersed. We're never just waiting for a non-empty batch though.

Reply:
that's why I wrote `if exists`. But ok, I will change it

Reply:
done